android/app/src/main/java/org/saltyrtc/demo/app/webrtc/PeerConnectionHelper.java

```diff
@@ -0,0 +1,389 @@
+/*
+ * Copyright (c) 2016-2019 Threema GmbH
+ *
+ * Licensed under the Apache License, Version 2.0, <see LICENSE-APACHE file>
+ * or the MIT license <see LICENSE-MIT file>, at your option. This file may not be
+ * copied, modified, or distributed except according to those terms.
+ */
+package org.saltyrtc.demo.app.webrtc;
+
+import android.support.annotation.AnyThread;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.saltyrtc.client.exceptions.ConnectionException;
+import org.saltyrtc.client.signaling.CloseCode;
+import org.saltyrtc.demo.app.BuildConfig;
+import org.saltyrtc.demo.app.Config;
+import org.saltyrtc.demo.app.MainActivity;
+import org.saltyrtc.demo.app.StateType;
+import org.saltyrtc.tasks.webrtc.WebRTCTask;
+import org.saltyrtc.tasks.webrtc.events.MessageHandler;
+import org.saltyrtc.tasks.webrtc.exceptions.UntiedException;
+import org.saltyrtc.tasks.webrtc.messages.Answer;
+import org.saltyrtc.tasks.webrtc.messages.Candidate;
+import org.saltyrtc.tasks.webrtc.messages.Offer;
+import org.saltyrtc.tasks.webrtc.transport.ISignalingTransportHandler;
+import org.saltyrtc.tasks.webrtc.transport.ISignalingTransportLink;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.webrtc.DataChannel;
+import org.webrtc.IceCandidate;
+import org.webrtc.MediaConstraints;
+import org.webrtc.MediaStream;
+import org.webrtc.PeerConnection;
+import org.webrtc.PeerConnectionFactory;
+import org.webrtc.RtpReceiver;
+import org.webrtc.SdpObserver;
+import org.webrtc.SessionDescription;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+@AnyThread
+public class PeerConnectionHelper {
```

Comment:
Again, mixed tabs and spaces. IIRC you can reformat the file with ctrl+alt+L or something like that.

There are also some other files that use spaces throughout.